Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -205,16 +212,47 @@ function getOrCreateSonarQubeStagingDirectory(): string {
     return sqStagingDir;
 }
 
+// Creates a string containing Markdown of the SonarQube quality gate status for this project.
+function createQualityGateStatusSection(taskReport: TaskReport): Q.Promise<string> {
+    if (!tl.getBoolInput('sqAnalysisWaitForAnalysis')) {
+        // Do not create a quality gate status section if not waiting for the server to analyse the build
+        console.log(tl.loc('sqCommon_NotWaitingForAnalysis'));
+        return Q.when<string>(null);
+    }
+
+    // Wait for analysis to finish so that we can get analysis information from the SonarQube server
+    var timeout:number = 60;
+    var delay:number = 1;
+
+    console.log(tl.loc('sqCommon_WaitingForAnalysis'));
+    return sqRest.waitForAnalysisCompletion(taskReport, timeout, delay)
+        .then((analysisCompleted:boolean) => {
+            if (!analysisCompleted) {
+                tl.debug('Did not receive a success response before the timeout expired.');
+                return Q.reject(tl.loc('sqAnalysis_AnalysisTimeout', timeout));
+            }
+
+            // If analysis completed, fetch the analysis ID to check quality gate status
+            return sqRest.getAnalysisId(taskReport)
+        })
+        .then((analysisId:string) => {
+            return sqRest.getAnalysisStatus(analysisId);
+        })
+        .then((analysisStatus:string) => {
+            return sqQualityGate.createBuildSummaryQualityGateSection(analysisStatus)
+        });
+}
+
 // Creates a string containing Markdown of a link to the SonarQube dashboard for this project.
-function createLinkToSonarQubeDashboard(sqBuildFolder: string): string {
-    var taskReport: TaskReport = getSonarQubeTaskReport(sqBuildFolder);
+function createLinkToSonarQubeDashboard(taskReport: TaskReport): string {
     if (!taskReport) {
+        // Looks like: Invalid or missing task report. Check SonarQube finished successfully.
         throw new Error(tl.loc('sqAnalysis_TaskReportInvalid'));
     }
 
-    return util.format('[%s >](%s "%s Dashboard")',
-        // Looks like: Detailed SonarQube report
-        tl.loc('sqAnalysis_BuildSummary_LinkText'), taskReport.dashboardUrl, taskReport.projectKey);
+    // Looks like: Detailed SonarQube report
```

Comment:
Uses of _util.format()_ have mostly been supplanted by the ES6 language-level feature, template strings.